internal/api/api_test.go

```diff
@@ -0,0 +1,133 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/kyma-project/btp-manager/internal/cluster-object/automock"
+	"github.com/kyma-project/btp-manager/internal/service-manager/automock"
+	"github.com/kyma-project/btp-manager/internal/service-manager/types"
+	"github.com/kyma-project/btp-manager/ui"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestApiResponses(t *testing.T) {
+
+	tests := []struct {
+		name           string
+		path           string
+		body           string
+		method         string
+		expectedStatus int
+		file           string
+		items          *types.ServiceInstances
+	}{
+		{
+			name:           "list instances should return all its services",
+			file:           "list-instances-happy-expected.json",
+			path:           "api/service-instances",
+			expectedStatus: http.StatusOK,
+			items: &types.ServiceInstances{
+				Items: []types.ServiceInstance{
+					{
+						Common: types.Common{
+							ID:          "1",
+							Name:        "service-1",
+							Description: "",
+						},
+					},
+					{
+						Common: types.Common{
+							ID:          "2",
+							Name:        "service-2",
+							Description: "",
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
```

Comment:
No test cases for unhappy path.

Reply:
This is simple API and tests are about to be refactored. Please propose test cases that you would like to see.

Reply:
`ListInstances` has two paths of execution. Happy (returning response and status 200) and unhappy (returning error response and 503. 
Test for unhappy flow would detect e.g. change in error status, and error handling in UI supposedly relies on some assumptions about statuses returned from API.

Reply:
Added 503 case in https://github.com/kyma-project/btp-manager/pull/746/commits/ca427c8fcabedcaafb278c93ffa44890afd900cc